packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -57,23 +69,49 @@ const hasNoAsyncOpOrExists = async (cumulusMessage, knex) => {
   }, knex, tableNames.asyncOperations);
 };
 
-const hasNoCollectionOrExists = async (cumulusMessage, knex) => {
-  const collectionInfo = getCollectionNameAndVersionFromMessage(cumulusMessage);
-  if (!collectionInfo) {
-    return true;
+const getMessageCollection = async (cumulusMessage, knex) => {
+  try {
+    const collectionNameAndVersion = getCollectionNameAndVersionFromMessage(cumulusMessage);
+    if (!collectionNameAndVersion) {
+      throw new Error('Could not find collection name/version in message');
+    }
+    return await knex(tableNames.collections).where(
+      collectionNameAndVersion
+    ).first();
+  } catch (error) {
+    log.error(error);
+    return undefined;
+  }
+};
+
+const getMessageProvider = async (cumulusMessage, knex) => {
+  try {
+    const providerId = getMessageProviderId(cumulusMessage);
+    if (!providerId) {
+      throw new Error('Could not find provider ID in message');
+    }
+    return await knex(tableNames.providers).where({
+      name: getMessageProviderId(cumulusMessage),
+    }).first();
+  } catch (error) {
+    log.error(error);
+    return undefined;
   }
-  return doesRecordExist(collectionInfo, knex, tableNames.collections);
 };
 
-const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
+const shouldWriteExecutionToRDS = async (
+  cumulusMessage,
+  collection,
+  knex
+) => {
   try {
-    const executionIsPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
-    if (!executionIsPostDeployment) return executionIsPostDeployment;
+    const isExecutionPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
+    if (!isExecutionPostDeployment) return false;
+    if (!isRecordDefined(collection)) return false;
 
     const results = await Promise.all([
       hasNoParentExecutionOrExists(cumulusMessage, knex),
```

Comment:
Not this PR so not blocking, but should these throw such that this doesn't await the other promise if there's a failure? 

That might also make checking if they're all true cleaner.